docs/src/man/hosting.md

```diff
@@ -97,15 +98,43 @@ Follow the instructions that are printed out, namely:
 
 ## `.travis.yml` Configuration
 
-In the `after_success` section of the `.travis.yml` file, where code coverage is processed,
-run your `docs/make.jl` file:
+To tell Travis that we want a new build stage we can add the following to the
+`.travis.yml` file:
 
 ```yaml
-after_success:
-  - julia -e 'Pkg.add("Documenter")'
-  - julia -e 'cd(Pkg.dir("PACKAGE_NAME")); include(joinpath("docs", "make.jl"))'
+jobs:
+  include:
+    - stage: "Documentation"
+      julia: 1.0
+      os: linux
+      script:
+        - julia --project=docs/ -e 'using Pkg; Pkg.instantiate()'
+        - julia --project=docs/ -e 'using Pkg; Pkg.add(PackageSpec(path=pwd()))'
+        - julia --project=docs/ docs/make.jl
+      after_success: skip
 ```
 
+where the `julia:` and `os:` entries decide the worker from which the docs
+are built and deployed. In the example above we will thus build and deploy
+the documentation from a linux worker running Julia 1.0.
+For more information on how to setup a build stage,
+see the Travis manual for [Build Stages](https://docs.travis-ci.com/user/build-stages).
+
+The three lines in the `script:` section does the following:
+ 1. Instantiate the doc-building environment (i.e. `docs/Project.toml`, see below).
+ 2. Install your package in the doc-build environment.
+ 3. Run the docs/make.jl script, which builds and deploys the documentation.
+
+The doc-build environment `docs/Project.toml` includes Documenter
+and other doc-build dependencies your package might have.
+If Documenter is the only dependency, then the `Project.toml`
+should include the following:
+```toml
+[deps]
+Documenter = "e30172f5-a6a5-5a46-863b-614d45cd2de4"
+```
```

Comment:
We should recommend that people have a `[compat]` section too here I think:

````
```toml
[deps]
Documenter = "e30172f5-a6a5-5a46-863b-614d45cd2de4"

[compat]
Documenter="0.20"
```
````

Reply:
added